flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -810,15 +812,24 @@ private void startHeartbeatServices() {
 			log);
 	}
 
-	private void assignScheduler(
+	private void reassignScheduler(
 			SchedulerNG newScheduler,
 			JobManagerJobMetricGroup newJobManagerJobMetricGroup) {
 		validateRunsInMainThread();
 		checkState(schedulerNG.requestJobStatus().isTerminalState());
 		checkState(jobManagerJobMetricGroup == null);
 
-		schedulerNG = newScheduler;
-		jobManagerJobMetricGroup = newJobManagerJobMetricGroup;
+		assignScheduler(newScheduler, newJobManagerJobMetricGroup);
+	}
+
+	private void assignScheduler(
+			SchedulerNG scheduler,
+			JobManagerJobMetricGroup jobManagerJobMetricGroup) {
+
+		this.schedulerNG = scheduler;
+		this.jobManagerJobMetricGroup = jobManagerJobMetricGroup;
+
+		log.info("Scheduler {} is used for job {}.", schedulerNG, jobGraph.getJobID());
```

Comment:
I wouldn't log this information here yet:
-  Scheduler configuration is per cluster at the moment. However, here it is logged per job.
- Potentially logged multiples times per job (if JobMaster loses and regains leadership)
- One can differentiate between the schedulers by looking at the cluster configuration, which is already being logged

Bottom line is that I would drop this commit.

Reply:
Agreed that we can infer from the cluster config log for the scheduler used. But it is not very straight forward, especially when it is not configured and the default config is used.
As we are already logging the failover strategy for each job, I think logging the scheduler for each job is acceptable.
To avoid logging it for multiple times, we can just log it the JobMaster constructor.

Reply:
You are right that in the absence of an explicit configuration entry/system variable, one has to check the config option's default value. However, there are still reasons against logging in the JobMaster constructor:
* Potentially logged multiple times in sesssion clusters.
* It is not expected that the end-user has to change the scheduler. Hopefully `legacy` will be removed soon and there are no concrete plans to include additional schedulers in the repository at the moment.
* No proper `toString()` implementation at the moment (classname@hexcode does not look nice).
* Side effects in constructor should be avoided.

Since the rest of the PR looks good, I will drop the commit for now.

Reply:
Fine.